From: Dan Carpenter <dan.carpenter@linaro.org>
Subject: mm,memory_hotplug: set failure reason in offline_pages()
Date: Wed, 25 Jun 2025 10:32:04 -0500

The "reason" variable is uninitialized on this error path.  It's supposed
to explain why the function failed.

Link: https://lkml.kernel.org/r/be4fd31b-7d09-46b0-8329-6d0464ffa7a5@sabinyo.mountain
Fixes: e4e2806b639c ("mm,memory_hotplug: implement numa node notifier")
Signed-off-by: Dan Carpenter <dan.carpenter@linaro.org>
Acked-by: David Hildenbrand <david@redhat.com>
Acked-by: Oscar Salvador <osalvador@suse.de>
Cc: Joanthan Cameron <Jonathan.Cameron@huawei.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory_hotplug.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/mm/memory_hotplug.c~mmmemory_hotplug-implement-numa-node-notifier-fix
+++ a/mm/memory_hotplug.c
@@ -1978,8 +1978,10 @@ int offline_pages(unsigned long start_pf
 		mem_arg.status_change_nid = node;
 		ret = node_notify(NODE_REMOVING_LAST_MEMORY, &node_arg);
 		ret = notifier_to_errno(ret);
-		if (ret)
+		if (ret) {
+			reason = "node notifier failure";
 			goto failed_removal_isolated;
+		}
 	}
 
 	ret = memory_notify(MEM_GOING_OFFLINE, &mem_arg);
_
